src/shared/sleep-config.c

```diff
@@ -102,6 +102,39 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+int battery_is_low(void) {
+        _cleanup_free_ char *bat_cap_level = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity_level", &bat_cap_level);
```

Comment:
hmm, so we cannot really hardcode this path, it might be different on different devices. "BAT0" is a chosen name by the acpi implementation, but could be anything else really, iiuc.

So, this should use the sd-device enumeration API, and then enumerate all devices of the "power_supply" subsystem, and ignore all devices which haven't got POWER_SUPPLY_TYPE=Battery set.

I also think the "capicty_level" is an optional property of devices of this types. At the very least we must handle its absence (i.e. inability to read it causing ENOENT here) gracefully and ignore the device (potentially, we need to find other ways, to implement this for devices lacking this property, but we can add this once we learn that there are relevant devices lacking it).

Also see ABI/testing/sysfs-class-power in the kernel tree.

Reply:
>> hmm, so we cannot really hardcode this path, it might be different on different devices. "BAT0" is a chosen name by the acpi implementation, but could be anything else really, iiuc.

will check how to do that